server/src/main/java/org/opensearch/wlm/tracker/ResourceUsageCalculator.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.tracker;
+
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.wlm.QueryGroupTask;
+
+import java.util.List;
+import java.util.function.LongSupplier;
+
+/**
+ * This class is used to track query group level resource usage
+ */
+@PublicApi(since = "2.18.0")
+public abstract class ResourceUsageCalculator {
+    /**
+     * calculates the current resource usage for the query group
+     *
+     * @param tasks        list of tasks in the query group
+     */
+    public abstract double calculateResourceUsage(List<QueryGroupTask> tasks);
+
+    /**
+     * calculates the task level resource usage
+     * @param task         QueryGroupTask
+     * @return task level resource usage
+     */
+    public abstract double calculateTaskResourceUsage(QueryGroupTask task);
+
+    /**
+     * Since only few implementations might need this
+     * @param nanoTimeSupplier
+     */
+    public void setNanoTimeSupplier(LongSupplier nanoTimeSupplier) {
+
+    }
```

Comment:
This does not look correct to me. I am wondering if we can leverage `ElapsedTimeTracker` or something similar for achieving this?

Reply:
I don;t think we can given the ElapsedTimeTracker works with time as long type while this feature as double. I didn't see any overlap. How do you propose we use that ?

Reply:
Moved this to QueryGroupTask.